react-game/src/hooks/useHttp.ts

```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+
+export const useHttp = () => {
+    const [loading, setLoading] = React.useState(false);
+    const [error, setError] = React.useState(null)
+
+    const request:any = React.useCallback(async (url: string, method = 'GET', body = null, headers = {}) => {
+        setLoading(true)
+        try {
+           const response =  await fetch(url, {method, body, headers})
+           const data = await response.json()
+           if (!response.ok) {
+               throw new Error(data.message || 'Somehting goes wrong')

```

Comment:
There are many spelling errors and typos in the application.
There's an extension, in VS Code, in case you are using it, that helps with the accidental typos (in this case there's a typo in Somehting, file game.ts - exapmle (instead of example), etc.)
Code Spell Checker - name of the extension.